azext_iot/tests/iothub/test_iot_ext_int.py

```diff
@@ -148,7 +228,110 @@ def test_identity_storage(self):
-                    LIVE_HUB, identity_type_disable
+                "iot hub identity remove -n {} --identities {}".format(
+                    LIVE_HUB, '[system]'
+                )
+            ).get_output_in_json()
+
+    @pytest.mark.skipif(
+        not all([LIVE_STORAGE_ID, LIVE_STORAGE]),
+        reason="azext_iot_identity_teststorageid and azext_iot_teststorageuri env vars not set",
+    )
+    @pytest.mark.skipif(
+        not ensure_iothub_sdk_min_version(IOTHUB_TRACK_2_SDK_MIN_VERSION),
+        reason="Skipping track 2 tests because SDK is track 1")
+    def test_user_identity_storage(self):
+        # User Assigned Managed Identity
+        storage_role = "Storage Blob Data Contributor"
+        user_identity = self.get_managed_identity()
+        identity_id = user_identity.get_output_in_json()["id"]
+        # check hub identity
+        identity_enabled = False
+
+        hub_identity = self.cmd(
+            "iot hub identity show -n {}".format(LIVE_HUB)
+        ).get_output_in_json()
+
+        if hub_identity.get("userAssignedIdentities", None) != user_identity.get_output_in_json()["principalId"]:
+            # enable hub identity and get ID
+            hub_identity = self.cmd(
+                "iot hub identity assign -n {} --user {}".format(
+                    LIVE_HUB, identity_id
+                )
+            )
+
+            identity_enabled = True
+
+        identity_principal = hub_identity["userAssignedIdentities"][identity_id]["principalId"]
+        assert identity_principal == user_identity.get_output_in_json()["principalId"]
+
+        # setup RBAC for storage account
+        storage_account_roles = self.cmd(
+            'role assignment list --scope "{}" --role "{}" --query "[].principalId"'.format(
+                LIVE_STORAGE_ID, storage_role
+            )
+        ).get_output_in_json()
+
+        if identity_principal not in storage_account_roles:
+            self.cmd(
+                'role assignment create --assignee "{}" --role "{}" --scope "{}"'.format(
+                    identity_principal, storage_role, LIVE_STORAGE_ID
                 )
             )
+            # give time to finish job
+            from time import sleep
+            sleep(60)
+
+        # identity-based device-identity export
+        self.cmd(
+            'iot hub device-identity export -n {} --bcu "{}" --auth-type {} --identity {} --ik true'.format(
+                LIVE_HUB, LIVE_STORAGE, "identity", identity_id
+            ),
+            checks=[
+                self.check("outputBlobContainerUri", LIVE_STORAGE),
+                self.check("failureReason", None),
+                self.check("type", "export"),
+                self.check("excludeKeysInExport", False),
+                self.check("storageAuthenticationType", "identityBased"),
+                self.exists("jobId"),
+            ],
+        )
+
+        # give time to finish job
+        from time import sleep
```

Comment:
The import for sleep is happening so many times might as well import once at the top of the module.